packages/Webkul/Product/src/Type/Booking.php

```diff
@@ -211,6 +211,13 @@ public function prepareForCart($data)
         $typeHelper = app($this->bookingHelper->getTypeHelper($bookingProduct->type));
 
         if (! $typeHelper->isSlotAvailable($products)) {
+            if ($bookingProduct->type == 'rental') {
+                return trans('shop::app.products.booking.cart.integrity.unavailable_date', [
+                    'date_from' => $data['booking']['date_from'] ?? null,
+                    'date_to'   => $data['booking']['date_to'] ?? null,
+                ]);
+            }
+
             return trans('shop::app.products.booking.cart.integrity.inventory_warning');
```

Comment:
I think we can simplify this by directly returning the message from the method, like so:

```php
return $typeHelper->slotUnavailableMessage($data);
```

We can define this method in the abstract booking type and then override it in the `RentalSlot`.

```php
/**
 * Return a slot unavailable message.
 */
public function slotUnavailableMessage(array $data): string
{
    ...
}
```